File: LLM-Translations/GPT-4/avatar/Java/Go/atcoder_ABC146_D.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type Edge struct {
	ID int
	To int
}

var scanner = bufio.NewScanner(os.Stdin)
var writer = bufio.NewWriter(os.Stdout)
var n int
var g = make([][]Edge, 0)
var ans []int

func main() {
	scanner.Scan()
	n, _ = strconv.Atoi(scanner.Text())
	ans = make([]int, n-1)
	for i := 0; i < n; i++ {
		g = append(g, make([]Edge, 0))
	}
	for i := 0; i < n-1; i++ {
		scanner.Scan()
		ab := strings.Split(scanner.Text(), " ")
		a, _ := strconv.Atoi(ab[0])
		b, _ := strconv.Atoi(ab[1])
		a--
		b--
		g[a] = append(g[a], Edge{i, b})
		g[b] = append(g[b], Edge{i, a})
	}
	dfs(0, -1, -1)
	max := 0
	for _, temp := range ans {
		max = maxInt(max, temp)
	}
	fmt.Println(max)
	for _, c := range ans {
		fmt.Println(c)
	}
}

func dfs(to int, color int, parents int) {
	k := 1
	for _, e := range g[to] {
		if e.To == parents {
			continue
		}
		if k == color {
			k++
		}
		ans[e.ID] = k
		dfs(e.To, k, to)
		k++
	}
}

func maxInt(a int, b int) int {
	if a > b {
		return a
	}
	return b
}

// //End of Code

File: LLM-Translations/GPT-4/avatar/Java/Python/atcoder_ABC146_D.py
Here is the Python translation of your Java code:


from collections import defaultdict

n = int(input())
g = defaultdict(list)
ans = [0]*(n-1)
edges = []

for i in range(n-1):
    a, b = map(int, input().split())
    a -= 1
    b -= 1
    g[a].append((i, b))
    g[b].append((i, a))
    edges.append((a, b))

def dfs(to, color, parents):
    k = 1
    for e in g[to]:
        if e[1] == parents:
            continue
        if k == color:
            k += 1
        ans[e[0]] = k
        dfs(e[1], k, to)
        k += 1

dfs(0, -1, -1)

print(max(ans))
for a, b in edges:
    print(ans[a, b])


